### PR TITLE
geometry cpp sample combining other shape detection samples

### PR DESCRIPTION
**Pull request by gursimarsingh**
Clean Samples #25006
This PR removes adds a new cpp sample (geometry) which combines different methods of finding and drawing shapes in an image. It makes separate samples for convexHull, fitellipse, minAreaRect, minAreaCircle redudant. Shapes can be changed using hotkeys after running the program

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: samples/cpp/geometry.cpp [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    "{help h||}{@image|ellipses.jpg|}{ellipse_method|0|Ellipse fitting method: 0 for OpenCV, 1 for AMS, 2 for Direct.}");
+
+    if (parser.has("help"))
+    {
+        help(argv);
+        return 0;
+    }
+
+    help(argv);
+    ellipseMethod = parser.get<int>("ellipse_method");
```
Comment (asmorkalov) [inline samples/cpp/geometry.cpp:75]: Range check is needed.

File: samples/cpp/geometry.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+inline static bool isGoodBox(const RotatedRect& box) {
+    //size.height >= size.width awalys,only if the pts are on a line or at the same point,size.width=0
+    return (box.size.height <= box.size.width * 30) && (box.size.width > 0);
+}
+
+// Function to draw fitted ellipses using different methods
+void drawFittedEllipses(Mat &img, vector<Point> &points) {
+    switch (ellipseMethod) {
+        case 0: // Standard ellipse fitting
```
Comment (asmorkalov) [inline samples/cpp/geometry.cpp:159]: I propose to introduce named constants.

File: samples/cpp/geometry.cpp [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+void drawMinAreaRect(Mat &img, vector<Point> &points);
+void drawFittedEllipses(Mat &img, vector<Point> &points);
+void drawMinEnclosingCircle(Mat &img, vector<Point> &points);
+void drawMinEnclosingTriangle(Mat &img, const vector<Point> &points);
+
+// Shape fitting options
+Mat image;
+int sliderPos = 70;
+string shape = "--all";
+int ellipseMethod = 0;
```
Comment (asmorkalov) [inline samples/cpp/geometry.cpp:61]: I propose to pass the value as function parameter in `drawFittedEllipses` and `drawShapes`

File: samples/cpp/geometry.cpp [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+int main(int argc, char** argv) {
+
+    cv::CommandLineParser parser(argc, argv,
+    "{help h||}{@image|ellipses.jpg|}{ellipse_method|0|Ellipse fitting method: 0 for OpenCV, 1 for AMS, 2 for Direct.}");
```
Comment (asmorkalov) [inline samples/cpp/geometry.cpp:66]: I propose to format command line parameters like in https://github.com/opencv/opencv/blob/b0d905864d6af8fef33ab30e102aa3cfcb329fe0/samples/cpp/aruco_dict_utils.cpp#L255

File: samples/cpp/geometry.cpp [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+}
+
+// Function to draw the minimum area rectangle around given points
+void drawMinAreaRect(Mat &img, vector<Point> &points) {
+    RotatedRect box = minAreaRect(points);  // Find the minimum area rectangle
+    Point2f vtx[4];
+    box.points(vtx);
+    // Draw the rectangle
+    for (int i = 0; i < 4; i++)
+        line(img, vtx[i], vtx[(i+1)%4], Scalar(0, 255, 0), 2, LINE_AA);
```
Comment (asmorkalov) [inline samples/cpp/geometry.cpp:140]: Looks like `polylines` can be used.

File: samples/cpp/geometry.cpp [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+    minEnclosingTriangle(points, triangle);  // Find the enclosing triangle
+
+    if (triangle.size() != 3) {
+        return;
+    }
+
+    // Draw the triangle
+    for (int i = 0; i < 3; i++) {
+        line(img, triangle[i], triangle[(i + 1) % 3], Scalar(255, 0, 0), 2, LINE_AA);
+    }
```
Comment (asmorkalov) [inline samples/cpp/geometry.cpp:130]: Looks like `polylines` can be used.

File: samples/cpp/geometry.cpp (hunk 1 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp:75]: @asmorkalov do you mean range check on number of input arguments or valid inputs of ellipse_method?

Comment (abhishekGola54) [reply samples/cpp/geometry.cpp]: @asmorkalov moved ellipseMethod to hot keys

File: samples/cpp/geometry.cpp (hunk 2 above)
Comment (abhishekGola54) [reply samples/cpp/geometry.cpp]: Done @asmorkalov

File: samples/cpp/geometry.cpp (hunk 1 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp]: @asmorkalov moved ellipseMethod to hot keys

File: samples/cpp/geometry.cpp (hunk 2 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp]: Done

File: samples/cpp/geometry.cpp (hunk 3 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp]: Done by using userData struct

File: samples/cpp/geometry.cpp (hunk 4 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp]: Done

File: samples/cpp/geometry.cpp (hunk 5 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp]: Done

File: samples/cpp/geometry.cpp (hunk 6 above)
Comment (gursimarsingh) [reply samples/cpp/geometry.cpp]: Done

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**